src/client/js/CoreEditor.js

```diff
@@ -376,14 +414,27 @@ class LinkUrlControls extends React.Component {
     }
     render() {
         return (
-            <div style={styles.urlInputContainer}>
+            <div style={styles.linkTitleInputContainer}>
+                <label>Title: </label>
+                <input
+                  onChange={this.onTitleChange}
+                  ref="linkTitle"
+                  style={styles.linkTitleInput}
+                  type="text"
+                  value={this.props.titleValue}
+                  onKeyDown={this.onTitleInputKeyDown}
+                />
+                <label>Url: </label>
+                <input
+                  ref="url"
+                  style={styles.linkURLInput}
+                  type="text"
+                />
+                <label>Description: </label>
                 <input
-                  onChange={this.onURLChange}
                   ref="url"
-                  style={styles.urlInput}
+                  style={styles.linkDescInput}
                   type="text"
-                  value={this.props.urlValue}
-                  onKeyDown={this.onLinkInputKeyDown}
                 />
                 <button onMouseDown={this.props.isEditLink ? this.onEditLink : this.onConfirmLink}>
```

Comment:
Please replace onMouseDown with onClick, it's more general and allows keyboard navigation.